<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "DTD/xhtml1-strict.dtd">
<html>
  

  <head>
    
    <title>sqlalchemy.engine.base.ExceptionContextImpl</title>
    <meta name="generator" content="pydoctor 22.2.0"> 
        
    </meta>
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=0.75" />
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
    <link rel="stylesheet" type="text/css" href="apidocs.css" />
    <link rel="stylesheet" type="text/css" href="extra.css" />
</head>

  <body>

    

    <nav class="navbar navbar-default">
  
  <div class="container">

    <div class="navbar-header">
      
      <div class="navlinks">
          <span class="navbar-brand">
            sqlalchemy <a href="index.html">API Documentation</a>
          </span>

          <a href="moduleIndex.html">
            Modules
          </a>

          <a href="classIndex.html">
            Classes
          </a>

          <a href="nameIndex.html">
            Names
          </a>
      </div>

    </div>
  </div>
</nav>

    

    <div class="container">

      <div class="page-header">
        <h1 class="class"><code><code><a href="index.html" class="internal-link">sqlalchemy</a></code><wbr></wbr>.<code><a href="sqlalchemy.engine.html" class="internal-link" title="sqlalchemy.engine">engine</a></code><wbr></wbr>.<code><a href="sqlalchemy.engine.base.html" class="internal-link" title="sqlalchemy.engine.base">base</a></code><wbr></wbr>.<code><a href="sqlalchemy.engine.base.ExceptionContextImpl.html" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl">ExceptionContextImpl</a></code></code></h1>
        <div id="showPrivate">
          <button class="btn btn-link" onclick="togglePrivate()">Toggle Private API</button>
        </div>
      </div>

      <div class="categoryHeader">
        class documentation
      </div>

      <div class="extrasDocstring">
        <p><code><span class="py-keyword">class</span> <span class="py-defname">ExceptionContextImpl</span>(<a href="sqlalchemy.engine.interfaces.ExceptionContext.html" class="internal-link" title="sqlalchemy.engine.interfaces.ExceptionContext">ExceptionContext</a>):</code></p>
        <p><a href="classIndex.html#sqlalchemy.engine.base.ExceptionContextImpl">View In Hierarchy</a></p>
      </div>

      <div class="moduleDocstring">
        <div>Implement the <code>.ExceptionContext</code> interface.</div>
      </div>

      <div id="splitTables">
        <table class="children sortable" id="id16114">
  
  
  <tr class="method">
    
    <td>Method</td>
    <td><code><a href="#__init__" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.__init__">__init__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#connection" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.connection">connection</a></code></td>
    <td>The <code>_engine.Connection</code> in use during the exception.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#engine" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.engine">engine</a></code></td>
    <td>The <code>_engine.Engine</code> in use during the exception.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#execution_context" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.execution_context">execution​_context</a></code></td>
    <td>The <code>.ExecutionContext</code> corresponding to the execution operation in progress.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#invalidate_pool_on_disconnect" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.invalidate_pool_on_disconnect">invalidate​_pool​_on​_disconnect</a></code></td>
    <td>Represent whether all connections in the pool should be invalidated when a "disconnect" condition is in effect.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#is_disconnect" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.is_disconnect">is​_disconnect</a></code></td>
    <td>Represent whether the exception as occurred represents a "disconnect" condition.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#original_exception" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.original_exception">original​_exception</a></code></td>
    <td>The exception object which was caught.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#parameters" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.parameters">parameters</a></code></td>
    <td>Parameter collection that was emitted directly to the DBAPI.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#sqlalchemy_exception" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.sqlalchemy_exception">sqlalchemy​_exception</a></code></td>
    <td>The <code><a href="sqlalchemy.exc.StatementError.html" class="internal-link">sqlalchemy.exc.StatementError</a></code> which wraps the original, and will be raised if exception handling is not circumvented by the event.</td>
  </tr><tr class="instancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="#statement" class="internal-link" title="sqlalchemy.engine.base.ExceptionContextImpl.statement">statement</a></code></td>
    <td>String SQL statement that was emitted directly to the DBAPI.</td>
  </tr>
</table>
        
          <p class="inheritedFrom">
            Inherited from <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html" class="internal-link" title="sqlalchemy.engine.interfaces.ExceptionContext">ExceptionContext</a></code>:
          </p>
          <table class="children sortable" id="id16115">
  
  
  <tr class="baseclassvariable">
    
    <td>Class Variable</td>
    <td><code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#chained_exception" class="internal-link" title="sqlalchemy.engine.interfaces.ExceptionContext.chained_exception">chained​_exception</a></code></td>
    <td>The exception that was returned by the previous handler in the exception chain, if any.</td>
  </tr><tr class="baseclassvariable">
    
    <td>Class Variable</td>
    <td><code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#cursor" class="internal-link" title="sqlalchemy.engine.interfaces.ExceptionContext.cursor">cursor</a></code></td>
    <td>The DBAPI cursor object.</td>
  </tr>
</table>
          

          
      </div>

      <div id="childList">

        <div class="basemethod">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.__init__">
    
  </a>
  <a name="__init__">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">__init__</span>(self, exception, sqlalchemy_exception, engine, connection, cursor, statement, parameters, context, is_disconnect, invalidate_pool_on_disconnect):
    
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.connection">
    
  </a>
  <a name="connection">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">connection</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#connection" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.connection</a></code></div>
    
    <div><p>The <code>_engine.Connection</code> in use during the exception.</p>
<p>This member is present, except in the case of a failure when
first connecting.</p>
<div class="rst-admonition seealso">
<p class="rst-first rst-admonition-title">See Also</p>
<p class="rst-last"><code>.ExceptionContext.engine</code></p>
</div>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.engine">
    
  </a>
  <a name="engine">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">engine</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#engine" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.engine</a></code></div>
    
    <div><p>The <code>_engine.Engine</code> in use during the exception.</p>
<p>This member should always be present, even in the case of a failure
when first connecting.</p>
<div class="rst-versionadded">
<span class="rst-versionmodified rst-added">New in version 1.0.0.</span></div>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.execution_context">
    
  </a>
  <a name="execution_context">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">execution_context</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#execution_context" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.execution_context</a></code></div>
    
    <div><p>The <code>.ExecutionContext</code> corresponding to the execution
operation in progress.</p>
<p>This is present for statement execution operations, but not for
operations such as transaction begin/end.  It also is not present when
the exception was raised before the <code>.ExecutionContext</code>
could be constructed.</p>
<p>Note that the <code>.ExceptionContext.statement</code> and
<code>.ExceptionContext.parameters</code> members may represent a
different value than that of the <code>.ExecutionContext</code>,
potentially in the case where a
<code>_events.ConnectionEvents.before_cursor_execute</code> event or similar
modified the statement/parameters to be sent.</p>
<p>May be None.</p>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.invalidate_pool_on_disconnect">
    
  </a>
  <a name="invalidate_pool_on_disconnect">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">invalidate_pool_on_disconnect</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#invalidate_pool_on_disconnect" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.invalidate_pool_on_disconnect</a></code></div>
    
    <div><p>Represent whether all connections in the pool should be invalidated
when a "disconnect" condition is in effect.</p>
<p>Setting this flag to False within the scope of the
<code>_events.ConnectionEvents.handle_error</code>
event will have the effect such
that the full collection of connections in the pool will not be
invalidated during a disconnect; only the current connection that is the
subject of the error will actually be invalidated.</p>
<p>The purpose of this flag is for custom disconnect-handling schemes where
the invalidation of other connections in the pool is to be performed
based on other conditions, or even on a per-connection basis.</p>
<div class="rst-versionadded">
<span class="rst-versionmodified rst-added">New in version 1.0.3.</span></div>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.is_disconnect">
    
  </a>
  <a name="is_disconnect">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">is_disconnect</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#is_disconnect" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.is_disconnect</a></code></div>
    
    <div><p>Represent whether the exception as occurred represents a "disconnect"
condition.</p>
<p>This flag will always be True or False within the scope of the
<code>_events.ConnectionEvents.handle_error</code> handler.</p>
<p>SQLAlchemy will defer to this flag in order to determine whether or not
the connection should be invalidated subsequently.    That is, by
assigning to this flag, a "disconnect" event which then results in
a connection and pool invalidation can be invoked or prevented by
changing this flag.</p>
<div class="rst-admonition note">
<p class="rst-first rst-admonition-title">Note</p>
<p class="rst-last">The pool "pre_ping" handler enabled using the
<a href="#system-message-1"><span class="rst-problematic" id="rst-problematic-1">:paramref:`_sa.create_engine.pool_pre_ping`</span></a> parameter does <strong>not</strong>
consult this event before deciding if the "ping" returned false,
as opposed to receiving an unhandled error.   For this use case, the
<a href="#system-message-2"><span class="rst-problematic" id="rst-problematic-2">:ref:`legacy recipe based on engine_connect() may be used
&lt;pool_disconnects_pessimistic_custom&gt;`</span></a>.  A future API allow more
comprehensive customization of the "disconnect" detection mechanism
across all functions.</p>
</div>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.original_exception">
    
  </a>
  <a name="original_exception">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">original_exception</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#original_exception" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.original_exception</a></code></div>
    
    <div><p>The exception object which was caught.</p>
<p>This member is always present.</p>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.parameters">
    
  </a>
  <a name="parameters">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">parameters</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#parameters" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.parameters</a></code></div>
    
    <div><p>Parameter collection that was emitted directly to the DBAPI.</p>
<p>May be None.</p>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.sqlalchemy_exception">
    
  </a>
  <a name="sqlalchemy_exception">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">sqlalchemy_exception</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#sqlalchemy_exception" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.sqlalchemy_exception</a></code></div>
    
    <div><p>The <code><a href="sqlalchemy.exc.StatementError.html" class="internal-link">sqlalchemy.exc.StatementError</a></code> which wraps the original,
and will be raised if exception handling is not circumvented by the event.</p>
<p>May be None, as not all exception types are wrapped by SQLAlchemy.
For DBAPI-level exceptions that subclass the dbapi's Error class, this
field will always be present.</p>
</div>
    
  </div>
</div><div class="baseinstancevariable">
  
  
  <a name="sqlalchemy.engine.base.ExceptionContextImpl.statement">
    
  </a>
  <a name="statement">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">statement</span> =
    
  </div>
  <div class="functionBody">
    <div class="interfaceinfo">overrides <code><a href="sqlalchemy.engine.interfaces.ExceptionContext.html#statement" class="internal-link">sqlalchemy.engine.interfaces.ExceptionContext.statement</a></code></div>
    
    <div><p>String SQL statement that was emitted directly to the DBAPI.</p>
<p>May be None.</p>
</div>
    
  </div>
</div>

      </div>
    </div>

    <footer class="navbar navbar-default">
  
  <div class="container">
    <a href="index.html">API Documentation</a> for sqlalchemy,
  generated by <a href="https://github.com/twisted/pydoctor/">pydoctor</a>
    22.2.0 at 2022-02-23 04:41:59.
  </div>
</footer>

    <script src="pydoctor.js" type="text/javascript"></script>

  </body>
</html>